#!/usr/bin/env python
PACKAGE = "navigation"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("lsq_center_weight",   double_t,    0, "Pulls path to the center of the corridor", 0,  0, 500)
gen.add("lsq_border_residual_at_border",   double_t,    0, "residual used to penalize the near to the border at the border", 0,  0, 500)
gen.add("lsq_border_start_distance",   double_t,    0, "start penalizing near to border at this distance", 0,  0, 0.5)
gen.add("lsq_prefered_point_weight",   double_t,    0, "Pulls path to the prefered point of the corridor", 0,  0, 500)
gen.add("lsq_distance_weight", double_t,    0, "Reduces the path length", 0,  0, 10000)
gen.add("lsq_triangle_area_weight", double_t,    0, "Reduces the path curvature", 0,  0, 500)
gen.add("lsq_curvature_weight",    double_t,    0, "Reduces the path curvature (deprecated)", 0,  0, 100)
gen.add("lsq_endpoint_angle_weight",    double_t,    0, "Reduces the path curvature", 0,  0, 500)
gen.add("lsq_center_value",    double_t,    0, "Defines parameter value of center", 0,  0, 1)
gen.add("ceres_max_num_iterations",    int_t,    0, "Maximum number of iterations in optimizer", 0,  0, 50)
gen.add("ceres_max_solver_time",    double_t,    0, "Maximum time for solving the optimization problem", 0,  0, 1)

exit(gen.generate(PACKAGE, "path_planning", "PathPlanning"))
 
